mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -89,7 +90,8 @@ private void copyHeaderFields(HeaderCollection headerCollection, ZonedDateTime i
     }
 
     private void copyMessageFields(MailboxMessage message, ZoneId zoneId) {
-        this.id = message.getUid();
+        this.id = message.getMessageId();
+        this.uid = message.getUid();
```

Comment:
So the id is unique and the uid is not unique. And the capabilities about unique ids (id if you follow me) is named Uid.
No sure we'll be able to read over this in a few months.